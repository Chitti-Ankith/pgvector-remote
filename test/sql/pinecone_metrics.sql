SET client_min_messages = 'notice';
CREATE TABLE t (val vector(3));
CREATE INDEX i ON t USING pinecone (val) WITH (metric = "invalid_metric");
CREATE INDEX i ON t USING pinecone (val) WITH (metric = "cosine");
CREATE INDEX i ON t USING pinecone (val vector_cosine_ops) WITH (metric = "euclidean");
