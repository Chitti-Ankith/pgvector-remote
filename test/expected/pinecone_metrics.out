SET client_min_messages = 'notice';
CREATE TABLE t (val vector(3));
CREATE INDEX i ON t USING pinecone (val) WITH (metric = "invalid_metric");
ERROR:  Invalid metric: invalid_metric
HINT:  Metric must be one of 'euclidean', 'cosine', 'dotproduct'
CREATE INDEX i ON t USING pinecone (val) WITH (metric = "cosine");
ERROR:  Invalid pinecone metric cosine for this opclass.
HINT:  Metric should be euclidean
CREATE INDEX i ON t USING pinecone (val vector_cosine_ops) WITH (metric = "euclidean");
ERROR:  Invalid pinecone metric euclidean for this opclass.
HINT:  Metric should be cosine
